src/utils/guildRole.ts

```diff
@@ -0,0 +1,53 @@
+import { INTERNAL_SERVER_ERROR } from "../constants/responses";
+import { DISCORD_BASE_URL } from "../constants/urls";
+import { env } from "../typeDefinitions/default.types";
+import {
+  createNewRole,
+  memberGroupRole,
+} from "../typeDefinitions/discordMessage.types";
+
+export async function createGuildRole(body: createNewRole, env: env) {
+  const createGuildRoleUrl = `${DISCORD_BASE_URL}/guilds/${env.DISCORD_GUILD_ID}/roles`;
+  const data = {
+    ...body,
+    name: body.rolename,
+  };
+  try {
+    const response = await fetch(createGuildRoleUrl, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+      body: JSON.stringify(data),
+    });
+    if (response.ok) {
+      return await response.json();
```

Comment:
Have a type here so it becomes easy to understand what will the JSON response look like

Reply:
added the type